event/event_target_bus.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+//go:build js
+
+package event
+
+import (
+	"sync"
+	"sync/atomic"
+	"syscall/js"
+
+	"github.com/sourcenetwork/goji"
+)
+
+type eventTargetSub struct {
+	id       uint64
+	listener js.Func
+	value    chan Message
+	events   []Name
+}
+
+// Message returns the message channel for the subscription.
+func (s *eventTargetSub) Message() <-chan Message {
+	return s.value
+}
+
+// eventTargetBus uses a JavaScript EventTarget to
+// manage subscribers and publish messages.
+type eventTargetBus struct {
+	target       goji.EventTargetValue
+	bufferSize   uint
+	subscriberID atomic.Uint64
+	subscribers  map[uint64]*eventTargetSub
+	closeMutex   sync.Mutex
+}
+
+// NewEventTargetBus creates a new bus from the given event target value.
```

Comment:
suggestion: Took me a few seconds to realise what `event target` meant here (I checked this before the documentation on the private type, which is also not going to be readily available to users).  I suggest tweaking this slightly to `given JavaScript EventTarget value.`

suggestion: The `js.Value` type does cover this slightly, but it might be worth expanding the documentation to describe when this bus should probably be used over the other bus (and I guess future buses).  It might be worth adding a similar line to `NewChannelBus` too.

nitpick: Whilst it is JS, it might be worth explicit in noting that it communicates across the runtime boundary using JSON.